modules/programs/nushell.nix

```diff
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.nushell;
+
+  configFile = config:
+    pkgs.runCommand "config.toml" {
+      buildInputs = [ pkgs.remarshal ];
+      preferLocalBuild = true;
+      allowSubstitutes = false;
+    } ''
+      remarshal -if json -of toml \
+        < ${pkgs.writeText "config.json" (builtins.toJSON config)} \
+        > $out
+    '';
+
+in {
+  meta.maintainers = [ maintainers.Philipp-M ];
+
+  options.programs.nushell = {
+    enable = mkEnableOption "nushell";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.nushell;
+      defaultText = literalExample "pkgs.nushell";
+      description = "The package to use for nushell.";
+    };
+
+    settings = mkOption {
+      type = with types;
+        let
+          prim = either bool (either int str);
```

Comment:
Isn't this the same as `oneOf [ bool int str ]` ?

Reply:
Thanks, I updated this with your suggestion.
To be honest, I copied a lot from a similar derivation (*starship*, which is used automatically by nushell btw.)